### PR TITLE
add uninstall playbook

### PR DESCRIPTION
**Pull request by philbrookes**
add uninstall playbook
few changes to fix bugs in the install playbook
renamed `install-all.yml` to `install.yml`
updated the readme.

Test:
1. run `ansible-playbook playbooks/install.yml -e eval_github_client_id=<client> -e eval_github_client_secret=<secret>`
2. run `ansible-playbook playbooks/uninstall.yml`
3. run `ansible-playbook playbooks/install.yml -e eval_github_client_id=<client> -e eval_github_client_secret=<secret>`

There should be no failed steps.

### CODE REVIEW

**Comment by david-martin** [comment]
@philbrookes Initial verification looks good.
Here's the output from watching projects during the uninstall

```
launcher             Terminating
managed-services-broker             Terminating
che                 Terminating
enmasse             Terminating
3scale              Terminating
sso                 Terminating
webapp              Terminating
managed-services-broker             
enmasse             
webapp              
launcher             Terminating
launcher             
sso                 
3scale              
che                 
```

I was able to successfully re-run the uninstall playbook multiple times without error, so good to have it idempotent.

I'll do a more thorough review of the scripts next before approving

File: README.md [hunk 1]
```diff
@@ -206,3 +206,11 @@ oc login https://<openshift-master-url>
 cd evals/
 ansible-playbook -i inventories/hosts playbooks/webapp.yml
 ```
+
+## Uninstallation steps
+
+Run the uninstall.yml playbook from inside the evals directory:
+```shell
+cd evals/
+ansible-playbook playbooks/uninstall.yml
```
Comment (david-martin) [inline README.md]: maybe add `-i inventories/hosts` to this example cmd too?

**Comment by david-martin** [comment]
@philbrookes After uninstall I tried to access the cluster, which redirects to sso still (this gives a 503).
Are we missing the removal of sso as the identity provider in openshift master-config.yml (and a restart of masters)?

File: evals/playbooks/uninstall.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      failed_when: output.stderr != '' and 'not found' not in output.stderr
+      changed_when: output.rc == 0
+    - 
+      name: Uninstall rhsso
+      include_role:
+        name: rhsso
+        tasks_from: uninstall
+      tags: ['rhsso']
+    - 
+      name: Delete the webapp
```
Comment (david-martin) [inline evals/playbooks/uninstall.yml]: small thing, but why 'Delete' here while other things say 'Uninstall'

**Comment by philbrookes** [comment]
> @philbrookes After uninstall I tried to access the cluster, which redirects to sso still (this gives a 503).
> Are we missing the removal of sso as the identity provider in openshift master-config.yml (and a restart of masters)?

Nice spot, I had overlooked that!

I've added code to fix that now.

File: evals/playbooks/uninstall.yml (hunk 2 above)
Comment (philbrookes) [reply evals/playbooks/uninstall.yml]: I've updated this to match the others now.

File: README.md (hunk 1 above)
Comment (philbrookes) [reply README.md]: good idea, updated.

File: evals/roles/che/tasks/uninstall.yml [hunk 3]
```diff
@@ -66,3 +70,4 @@
     status_code: [204]
   register: che_clients_cmd
   when: che_keycloak_auth_response.status != 503
+  when: delete_keycloak_user
```
Comment (david-martin) [inline evals/roles/che/tasks/uninstall.yml]: @philbrookes 
This change gives a warning. We probably want to combine these 2?

```
 [WARNING]: While constructing a mapping from
/root/installation/evals/roles/che/tasks/uninstall.yml, line 62, column 3, found a
duplicate dict key (when). Using last defined value only.
```

Comment (philbrookes) [reply evals/roles/che/tasks/uninstall.yml]: whoops, changed this to an 'and'.

**Pull request merged**